### PR TITLE
Chasm Hell On Icebox - 300 Active Turfs on Prod Moment

### PR DESCRIPTION
**Pull request by san7890**
## About The Pull Request

Spontaneous regressions introduced by #74359 (1e58c1875d9e2f48a306fe31a0626dbbb1990ff9).
```txt
 - Z-Level 2 has 150 active turf(s).
 - Z-Level 3 has 150 active turf(s).
 - Z-Level trait Ice Ruins Underground has 300 active turf(s).
 - Z-Level trait Mining has 300 active turf(s).
 - Z-Level trait Station has 300 active turf(s).
 - End of active turf list.
 ```

image

Basically the lavaland ruin sucks dogshit and I had to do a lot of stuff to account for everything failing. There was even a moment where we were adding something to `flags_1` instead of `turf_flags` and that was also really bad to figure out.

image

i also had to add orange genturfs because it was really getting bad with all of the assertions we had to keep making, especially since stuff like this could also show up:

image

That's the prison in the red box, those are active turfs because a chasm scraped it away.

Sorry if this is hard to follow but I promise you everything in this is essential. I wish we didn't have to rely on turf flags as much as we do but this is a fix PR, not a refactor.
## Why It's Good For The Game

Even one active turf on IceBox ate up _three_ seconds of SSair's initialization every single time it was really fucking bad.

We haven't had to deal with chasms for about two years so there's a lot of mapping assertions we made since they just weren't a thing, but now they're back so lets do it properly.
## Changelog
:cl:
fix: The prison on IceBox should no longer leak air as often.
/:cl:

I have compiled this map about 30 times until active turfs stopped fucking happening and now I am content. This likely doesn't fix _everything_ because some stuff can still be hidden to me, and we still have PRs that need to be merged to reduce the amount of noise we're getting on prod.

### CODE REVIEW

**Comment by TheVekter** [comment]
> Even one active turf on IceBox ate up three seconds of SSair's initialization every single time it was really fucking bad.

Fuck me. Is this why Icebox got so insanely laggy in long rounds?

File: code/datums/mapgen/JungleGenerator.dm [hunk 1]
```diff
@@ -79,22 +79,27 @@
...
 	map_generator = /datum/map_generator/jungle_generator
 
+/// This turf doesn't actually do anything beyond provide contrast for mappers and be very visible when stuff breaks in game. The actual areas are what drive cave generation.
+/turf/open/genturf
+	name = "green ungenerated turf"
+	desc = "If you see this, and you're not a ghost, yell at coders"
+	icon = 'icons/turf/debug.dmi'
+	icon_state = "genturf_green"
+
+// following two are currently used for edge cases in which you want a certain type of map generation intermingled with other genturfs
```
[the remark this answers is not in the document]
Comment (san7890) [reply code/datums/mapgen/JungleGenerator.dm:96]: Yes, the genturfs are only rver there to rapidly spot if stuff is broken as well as be clean contrast. The area is the only thing that ever matters.

**Comment by san7890** [comment]
> > Even one active turf on IceBox ate up three seconds of SSair's initialization every single time it was really fucking bad.
> 
> 
> 
> Fuck me. Is this why Icebox got so insanely laggy in long rounds?

Did this occur before this previous Thursday? It would be something else if so. This just cuts down on initialize, though it's likely we may have been losing CPU time processing needlessly stimulated turfs

**Comment by EOBGames** [comment]
yeah, the map aspect all looks okay, provided I'm understanding the orange turf correctly

**Review by EOBGames — APPROVED** [review]
